### PR TITLE
[runtime] Replace pedump boehm dependency with sgen dependency

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

**Comment by alexanderkyte** [comment]
@directhex it built locally for me and pedump ran on my whole folder of bug repro .exe files without a crash. Should be good, but let's watch the CI.

File: mono/metadata/Makefile.am [hunk 1]
```diff
@@ -301,27 +299,25 @@ if DTRACE_G_REQUIRED
 
 PEDUMP_DTRACE_OBJECT = pedump-dtrace.$(OBJEXT)
 
-pedump-dtrace.$(OBJEXT): $(top_srcdir)/data/mono.d libmonoruntime.la ../io-layer/libwapi.la ../utils/libmonoutils.la
+pedump-dtrace.$(OBJEXT): $(top_srcdir)/data/mono.d $(sgen_libraries) ../io-layer/libwapi.la ../utils/libmonoutils.la
 	DTRACE="$(DTRACE)" DTRACEFLAGS="$(DTRACEFLAGS)" AR="$(AR)" $(SHELL) $(top_srcdir)/data/dtrace-prelink.sh \
-	--pic pedump-dtrace.$(OBJEXT) $(top_srcdir)/data/mono.d libmonoruntime.la ../io-layer/libwapi.la ../utils/libmonoutils.la
+	--pic pedump-dtrace.$(OBJEXT) $(top_srcdir)/data/mono.d $(sgen_libraries) ../io-layer/libwapi.la ../utils/libmonoutils.la
```
Comment (vargaz) [inline mono/metadata/Makefile.am]: Shouldn't this be libmonoruntimesgen.la ?

File: mono/metadata/Makefile.am [hunk 2]
```diff
@@ -33,10 +33,8 @@ platform_sources += ../../support/libm/complex.c
 endif
 
 if SHARED_MONO
-if SUPPORT_BOEHM
 bin_PROGRAMS = pedump
 endif
-endif
```
Comment (vargaz) [inline mono/metadata/Makefile.am]: This should probably be SUPPORT_SGEN.

File: mono/metadata/Makefile.am (hunk 1 above)
Comment (alexanderkyte) [inline mono/metadata/Makefile.am]: Ah right. I'll make this shared_sgen_libraries instead.

**Comment by akoeplinger** [comment]
There's a build error now:

```
make[3]: *** No rule to make target `../sgen/libmonosgen.la', needed by `pedump'.  Stop.
00:07:29 make[3]: Leaving directory `/mnt/jenkins/workspace/test-mono-pull-request-amd64/mono/metadata'
00:07:29 
make[2]: *** [all-recursive] Error 1
```

Would be cool to get this in since afaik it's a precursor to switching off boehm for the Windows x64 builds (which we need to fix them), right?

**Comment by alexanderkyte** [comment]
Just needed to make an explicit cross-Makefile dependency.

**Comment by lewurm** [comment]
this change broke `mono-master-monodroid`:

```
[...]
[6:29:41] clang38: warning: -Wl,-z,relro: 'linker' input unused
[6:29:41] clang38: warning: -Wl,-z,noexecstack: 'linker' input unused
[6:29:41] make -w -C /Users/builder/data/lanes/1845/f1dd6fe7/source/mono/mono/sgen libmonosgen.la
[6:29:41] make[7]: Entering directory `/Users/builder/data/lanes/1845/f1dd6fe7/source/mono/mono/sgen'
[6:29:41] make[7]: Leaving directory `/Users/builder/data/lanes/1845/f1dd6fe7/source/mono/mono/sgen'
[6:29:41] make[7]: *** No rule to make target `libmonosgen.la'.  Stop.
[6:29:41] make[6]: *** [/Users/builder/data/lanes/1845/f1dd6fe7/source/mono/mono/sgen/libmonosgen.la] Error 2
[6:29:41] make[6]: *** Waiting for unfinished jobs....
[6:29:41] clang38: warning: -Wl,-z,now: 'linker' input unused
[6:29:41] clang38: warning: -Wl,-z,relro: 'linker' input unused
[6:29:41] clang38: warning: -Wl,-z,noexecstack: 'linker' input unused
[...]
```
@alexanderkyte: can you have a look please?

**Comment by vargaz** [comment]
How about we disable pedump and/or move it to tools/? Is anybody really using it ?

**Pull request merged**